shell/platform/fuchsia/flutter/platform_view_unittest.cc

```diff
@@ -528,30 +469,34 @@ TEST_F(PlatformViewTests, GetGrContextTest) {
                            nullptr                               // io
       );
 
-  // Test get GrContext callback function.
-  sk_sp<GrDirectContext> gr_context =
+  // Test create surface callback function.
+  MockExternalViewEmbedder mock_external_view_embedder;
+  sk_sp<GrDirectContext> mock_gr_context =
       GrDirectContext::MakeMock(nullptr, GrContextOptions());
-  auto GetGrContextCallback = [gr_context = gr_context.get()]() {
-    return gr_context;
-  };
+  std::unique_ptr<flutter::Surface> surface =
+      std::make_unique<flutter_runner::Surface>("CreateSurfaceTest",
+                                                &mock_external_view_embedder,
+                                                mock_gr_context.get());
+  flutter::Surface* surface_ptr = surface.get();
+  auto CreateSurfaceCallback = [surface = std::move(surface)]() mutable
+      -> std::unique_ptr<flutter::Surface> { return std::move(surface); };
 
   auto platform_view = flutter_runner::PlatformView(
       delegate,                               // delegate
       "test_platform_view",                   // label
       std::move(view_ref),                    // view_refs
       std::move(task_runners),                // task_runners
       services_provider.service_directory(),  // runner_services
-      nullptr,               // parent_environment_service_provider_handle
-      nullptr,               // session_listener_request
-      nullptr,               // focuser
-      nullptr,               // on_session_listener_error_callback
-      nullptr,               // on_enable_wireframe_callback,
-      nullptr,               // on_create_view_callback,
-      nullptr,               // on_destroy_view_callback,
-      nullptr,               // on_get_view_embedder_callback,
-      GetGrContextCallback,  // on_get_gr_context_callback,
-      0u,                    // vsync_event_handle
-      {}                     // product_config
+      nullptr,  // parent_environment_service_provider_handle
```

Comment:
FYI, this style of argument comments is non-idomatic for the c++ style ( https://google.github.io/styleguide/cppguide.html#Function_Argument_Comments ).  If you name them idiomatically you get linter checks making sure they are correct.  (I know they are wrong all over the place).

Reply:
Agreed, I'd like to go clean them out of the fuchsia shell in one pass when im done with this migration